src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -63,20 +67,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(lectureEntity);
-
-    // TODO: Add redirect link to lecture site.
+    response.sendRedirect("/index.html");
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.sendError(500, "Not Implemented");
+    Query query = new Query(LECTURE);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Lecture> lectures = new ArrayList<>();
+    for (Entity lecture : results.asIterable()) {
+      long id = lecture.getKey().getId();
+      String lectureName = (String) lecture.getProperty(LECTURE_NAME);
+      String videoUrl = (String) lecture.getProperty(VIDEO_URL);
+      String videoId = (String) lecture.getProperty(VIDEO_ID);
+
+      // Creates new Comment for JSON accessibility
+      Lecture newLecture = Lecture.create(id, lectureName, videoUrl, videoId);
+      lectures.add(newLecture);
+    }
+
+    Gson gson = new Gson();
+    response.setContentType("text/html");
+    response.getWriter().println(gson.toJson(lectures));
   }
 
   /**
-   * Returns value with {@code name} from the {@code request} form. 
+   * Returns value with {@code name} from the {@code request} form.
    * If the {@code name} cannot be found, return {@code defaultValue}.
    * @param request Form sent by client
```

Comment:
Newline before block tags

```suggestion
   *
   * @param request Form sent by client
```

Reply:
removed